### PR TITLE
Add refactoring code

### PR DESCRIPTION
**Pull request by vladeugeniu**

### CODE REVIEW

File: python/fazan/fazan.py [hunk 1]
```diff
@@ -34,10 +34,12 @@ def __init__(self):
      def my_turn(self,pref):
         f = open(pref + '.txt','r+')
         s = 0
-        lines = []
-        for line in f:
-            lines.append(line)
+        #lines = f.readlines()
```
Comment (razvand) [inline python/fazan/fazan.py]: Remove commented line.

File: python/fazan/fazan.py [hunk 2]
```diff
@@ -47,32 +49,26 @@ def my_turn(self,pref):
...
 
     print "Prefixul pentru cuvantul tau este: ", pref
     i = 0
 
     while i < 3:
         alegere = player.my_turn();
-        alegere = alegere + '\n'
-
-
+
```
Comment (razvand) [inline python/fazan/fazan.py:43]: Remove blank line.

File: python/fazan/fazan.py [hunk 3]
```diff
@@ -122,7 +119,7 @@ def computer_turn(pref,computer):
            print 'Cuvantul curetnt:',cuv_curent
            cuv_curent = player_turn(cuv_curent[len(cuv_curent)-3:len(cuv_curent)],player)
            if cuv_curent != False:
-                   cuv_curent = computer_turn(cuv_curent[len(cuv_curent)-3:len(cuv_curent)],computer)
+                   cuv_curent = computer_turn(cuv_curent[len(cuv_curent)-2:len(cuv_curent)],computer)
```
Comment (razvand) [inline python/fazan/fazan.py]: Add comment to argument the use of len(cuv_curent)-2. Why -2.

File: python/fazan/fazan.py [hunk 4]
```diff
@@ -122,7 +119,7 @@ def computer_turn(pref,computer):
            print 'Cuvantul curetnt:',cuv_curent
            cuv_curent = player_turn(cuv_curent[len(cuv_curent)-3:len(cuv_curent)],player)
```
Comment (razvand) [inline python/fazan/fazan.py]: Add comment to argument the use of len(cuv_curent)-3. Why -3.

File: python/fazan/partitionare.py [hunk 5]
```diff
@@ -22,3 +21,6 @@
 exi = raw_input("EXIIIIT")
 
 #input("fail ")
+T")
+
+#input("fail ")
```
Comment (razvand) [inline python/fazan/partitionare.py]: Remove commented line.

File: python/fazan/fazan.py [hunk 6]
```diff
@@ -32,92 +32,68 @@ class Calculator(Player):
...
-        print "Ai fost incuiat, nu exista niciun cuvant care incepe cu", pref
-        return False
-    lines = [line.rstrip('\n') for line in f]
 
-    print "Prefixul pentru cuvantul tau este: ", pref
-    i = 0
 
-    while i < 3:
-        alegere = player.my_turn();
+files_in_dir = os.listdir('.')
```
Comment (razvand) [inline python/fazan/fazan.py]: Add comments for files_in_dir and word_list global variables.

File: python/fazan/fazan.py [hunk 7]
```diff
@@ -32,92 +32,68 @@ class Calculator(Player):
...
-    g.seek(0)
-    rand = random.randrange(len(lines))
-    cuv_curent = lines[rand]
-    del lines[rand]
-    for line in lines:
-        g.write(line)
-
-    while True:
+     
+     rand1 = random.choice(word_list.keys())
```
Comment (razvand) [inline python/fazan/fazan.py]: Why name this rand1? Why not use random_word? Or cuv_aleator.

File: python/fazan/fazan.py [hunk 8]
```diff
@@ -32,92 +32,68 @@ class Calculator(Player):
...
+     cuv_curent = random.choice(word_list[rand1])
+     
+
```
Comment (razvand) [inline python/fazan/fazan.py]: Remove two out of three blank lines.

File: python/fazan/fazan.py [hunk 9]
```diff
@@ -129,5 +105,5 @@ def computer_turn(pref,computer):
                player.add_faz();
                break
 
-    print player.nume + ':', player.fazan
-    print computer.nume + ':', computer.fazan
+     print player.nume + ':', player.fazan
```
Comment (razvand) [inline python/fazan/fazan.py:99]: Extra space in indentation.

Comment (vladeugeniu) [inline python/fazan/fazan.py:99]: already fixed

File: python/fazan/fazan.py (hunk 4 above)
Comment (vladeugeniu) [inline python/fazan/fazan.py]: Actually it is -2, it was -3 before .rstrip('\n')

**Comment by vladeugeniu** [comment]
@razvand Done

File: python/fazan/fazan.py [hunk 10]
```diff
@@ -27,102 +28,65 @@ def add_faz(self):
...
-        del lines[rand]
-        for line in lines:
-            f.write(line)
-        f.close()
+        global word_list
+        cuv = random.choice(word_list[pref])
+        word_list[pref].remove(cuv)
         return cuv
 
+files_in_dir = os.listdir('.')   #a list containing the names of the entries in the current directory
```
Comment (razvand) [inline python/fazan/fazan.py:40]: Use proper commenting style: make it a sentence, start with Capitalized word, end with dot. Leave space between # and comment.

File: python/fazan/fazan.py [hunk 11]
```diff
@@ -27,102 +28,65 @@ def add_faz(self):
...
+word_list  = {} #a dictionary containing the words categorized by their prefixes
+for file in files_in_dir:
+ 
+    if '.txt' in file:
```
Comment (razvand) [inline python/fazan/fazan.py:44]: This means that "a.txt.akakakaka.sakjksajsa" matches. I'm not sure that's what you want.

File: python/fazan/fazan.py [hunk 12]
```diff
@@ -27,102 +28,65 @@ def add_faz(self):
...
+        print "Prefixul pentru cuvantul tau este: ", pref
+        i = 0
+        while i < 3:
+            alegere = player.my_turn();
+            if alegere in word_list[pref]:
+                word_list[pref].remove(alegere)
+                break
+            i += 1           
+        else:
+            print "Ai esuat in a gasi un cuvant, ai fost inchs! "
```
Comment (razvand) [inline python/fazan/fazan.py:61]: s/inchs/inchis/g

**Pull request merged**